TheMET/TheMET/TagListView/TagListView.swift

```diff
@@ -0,0 +1,205 @@
+                                      size: newTagView.intrinsicContentSize)
+            newTagView.translatesAutoresizingMaskIntoConstraints = true
+            newTagView.autoresizingMask = [ .flexibleRightMargin, .flexibleBottomMargin ]
+        }
+        self.tagViews = newTagViews
+        self.layoutTagViews()
+    }
+    
+    private func layoutTagViews() {
+        let visibleTags:[CGRect] = self.calculateLayout(for: self.bounds.width).tagViewFrames
+        let visibleTagsCount = visibleTags.count
+    throughTags: for (tagViewIndex, tagView) in tagViews.enumerated() {
+        if tagViewIndex < visibleTagsCount {
+            self.addSubview(tagView)
+            tagView.apply(radius: 15, backgroundColor: UIColor(named: "plum"), fontColor: UIColor(named: "blackberry"), font: NSLocalizedString("san_serif_font", comment: ""), fontSize: 13, buttonTitle: (self.tagViews[tagViewIndex].titleLabel?.text)!)
+            let tagFrameOrigin = self.calculateLayout(for: self.bounds.width).tagViewFrames[tagViewIndex].origin
+            tagView.frame.origin = tagFrameOrigin
+        }
+    }
+        if let showMoreButtonFrame = self.calculateLayout(for: self.bounds.width).showMoreButtonFrame,
+           showMoreButtonFrame.width != 0 {
+            self.addSubview(self.showMoreTagsButton)
+            self.showMoreTagsButton.frame.origin = showMoreButtonFrame.origin
+        }
+        self.invalidateIntrinsicContentSize()
+    }
+    
+    private func tagsWidth(from index1: Int, to index2: Int, xSpace: CGFloat) -> CGFloat {
+        var result: CGFloat = 0
+        for index in index1 ... index2 {
+            let tagView = self.tagViews[index]
+            result += tagView.bounds.width
+        }
+        result += xSpace * CGFloat(index2 - index1) + self.tagViews[index1 - 1].bounds.width
+        return result
+    }
+    
+    private func setRestTagsFrames(from index1: Int, to index2: Int, xSpace: CGFloat, ySpace: CGFloat, x: CGFloat, y: CGFloat) -> [CGRect] {
+        var tagViewFrames: [CGRect] = []
+        var xCoord: CGFloat = x
+        for index in index1 ... index2 {
+            let tagView = self.tagViews[index]
+            tagViewFrames.append(CGRect(x: xCoord, y: y, width: tagView.bounds.width, height: tagView.bounds.height))
+            xCoord += tagView.bounds.width + xSpace
+        }
+        return tagViewFrames
+    }
+    
+    private func setRestTagsAndShowMoreButtonFrames(from index1: Int, to index2: Int, xSpace: CGFloat, ySpace: CGFloat, x: CGFloat, y: CGFloat) -> (tagViewFrames: [CGRect], showMoreButton: CGRect) {
+        var result: ([CGRect], CGRect) = ([], .zero)
+        var xCoord: CGFloat = x
+        for index in index1 ... index2 {
+            let tagView = self.tagViews[index]
+            if tagView.bounds.width > self.bounds.width - xCoord - self.showMoreTagsButton.bounds.width {
+                self.showMoreTagsButton.frame.origin = CGPoint(x: xCoord, y: y)
```

Comment:
This method must not set the origin of the `self.showMoreTagsButton`, it should only calculate a `showMoreButtonFrame`